packages/frontend/core/src/components/pure/trash-page-footer/index.tsx

```diff
@@ -0,0 +1,118 @@
+import { Button } from '@affine/component/ui/button';
+import { ConfirmModal } from '@affine/component/ui/modal';
+import { Tooltip } from '@affine/component/ui/tooltip';
+import { WorkspaceSubPath } from '@affine/env/workspace';
+import { useAFFiNEI18N } from '@affine/i18n/hooks';
+import { assertExists } from '@blocksuite/global/utils';
+import { DeleteIcon, ResetIcon } from '@blocksuite/icons';
+import { useBlockSuitePageMeta } from '@toeverything/hooks/use-block-suite-page-meta';
+import { useCallback, useRef, useState } from 'react';
+
+import { useAppSettingHelper } from '../../../hooks/affine/use-app-setting-helper';
+import { useBlockSuiteMetaHelper } from '../../../hooks/affine/use-block-suite-meta-helper';
+import { useCurrentWorkspace } from '../../../hooks/current/use-current-workspace';
+import { useNavigateHelper } from '../../../hooks/use-navigate-helper';
+import { toast } from '../../../utils';
+import * as styles from './styles.css';
+
+export const TrashPageFooter = ({ pageId }: { pageId: string }) => {
+  // fixme(himself65): remove these hooks ASAP
+  const [workspace] = useCurrentWorkspace();
+  assertExists(workspace);
+  const blockSuiteWorkspace = workspace.blockSuiteWorkspace;
+  const pageMeta = useBlockSuitePageMeta(blockSuiteWorkspace).find(
+    meta => meta.id === pageId
+  );
+  assertExists(pageMeta);
+  const t = useAFFiNEI18N();
+  const { appSettings } = useAppSettingHelper();
+  const { jumpToSubPath } = useNavigateHelper();
+  const { restoreFromTrash } = useBlockSuiteMetaHelper(blockSuiteWorkspace);
+  const restoreRef = useRef(null);
+  const deleteRef = useRef(null);
+  const hintTextRef = useRef(null);
+  const [open, setOpen] = useState(false);
+  const hintText =
+    'This page has been moved to the trash, you can either restore or permanently delete it.';
+
+  return (
+    <div
+      className={styles.deleteHintContainer}
+      data-has-background={!appSettings.clientBorder}
+    >
+      <Tooltip
+        content={hintText}
+        portalOptions={{
+          container: hintTextRef.current,
+        }}
+        options={{ style: { whiteSpace: 'break-spaces' } }}
+      >
+        <div ref={hintTextRef} className={styles.deleteHintText}>
+          {hintText}
+        </div>
+      </Tooltip>
+
+      <div className={styles.group}>
+        <Tooltip
+          content={t['com.affine.trashOperation.restoreIt']()}
+          portalOptions={{
```

Comment:
when first render, the ref is null. I am curious why this ref is needed.

Reply:
Oh, this file is moved and I haven't checked the implementation. Let me take a look

Reply:
the ref is not uncessary. Removed.